tests/v2/validation/rbac/globalrolesv2/globalroles_v2.go

```diff
@@ -0,0 +1,189 @@
+package globalrolesv2
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	v1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/rancher/rancher/tests/framework/extensions/kubeapi/rbac"
+	"github.com/rancher/rancher/tests/framework/extensions/provisioning"
+	"github.com/rancher/rancher/tests/framework/extensions/provisioninginput"
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+)
+
+const (
+	roleOwner          = "cluster-owner"
+	roleMember         = "cluster-member"
+	roleCrtbView       = "clusterroletemplatebindings-view"
+	roleProjectsCreate = "projects-create"
+	roleProjectsView   = "projects-view"
+	standardUser       = "user"
+	localcluster       = "local"
+	crtbOwnerLabel     = "authz.management.cattle.io/grb-owner"
+	namespace          = "fleet-default"
+)
+
+var globalRole = v3.GlobalRole{
+	ObjectMeta: metav1.ObjectMeta{
+		Name: "",
+	},
+	InheritedClusterRoles: []string{},
+}
+
+var globalRoleBinding = &v3.GlobalRoleBinding{
+	ObjectMeta: metav1.ObjectMeta{
+		Name: "",
+	},
+	GlobalRoleName: "",
+	UserName:       "",
+}
+
+// Private helper function - Retrieves the Global Role Binding for a specific user.
+func getGlobalRoleBindingForUser(client *rancher.Client, userID string) (string, error) {
+	grblist, err := rbac.ListGlobalRoleBindings(client, localcluster, metav1.ListOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	for _, grbs := range grblist.Items {
+		if grbs.GlobalRoleName == globalRole.Name && grbs.UserName == userID {
+			return grbs.Name, nil
+		}
+	}
+	return "", nil
+}
+
+// Private helper function - List Cluster Role Template Bindings associated with a specific global role binding.
```

Comment:
Please remove the comment here and from the other places. We do not need to add comments for helper functions.